.github/workflows/variables.yml

```diff
@@ -41,13 +44,15 @@ jobs:
       release: ${{steps.parse.outputs.release}}
       patch: ${{steps.parse.outputs.patch}}
       rc: ${{steps.parse.outputs.rc}}
-      milestone: ${{steps.parse.outputs.milestone}}
-      next-milestone: ${{steps.parse.outputs.next-milestone}}
+      name: ${{steps.parse.outputs.name}}
+      dash-name: ${{steps.parse.outputs.dash-name}}
+      next-rc: ${{steps.parse.outputs.next-milestone}}
     steps:
       - id: parse
         run: |
-          read RELEASE PATCH_NUMBER RC_NUMBER <<< \
-          $(sed -rn 's/^([0-9]+\.[0-9]+)\.([0-9]+)(-rc\.([0-9]+))?$/\1 \2 \4/p' <<< ${{inputs.version}})
+          set -u
+          read RELEASE PATCH_NUMBER RELEASE_NAME RC_NUMBER <<< \
+          $(sed -rn 's/^([0-9]+\.[0-9]+)\.([0-9]+)(-([a-zA-Z0-9-]+))?(-rc\.([0-9]+))?$/\1 \2 \4 \6/p' <<< ${{inputs.version}})
```

Comment:
I am not fully sure about this group: `(-([a-zA-Z0-9-]+))?`

It would allow such things as: `x.y.z---rc.0`

How about we allow only valid kebab-case strings (that start with `-`) using: `(-([a-z][a-z0-9]*)(-[a-z0-9]+)*)`? 
(capitals can be optional - I think, we do not need them.)

Reply:
Well, that's possible, why not. Although the "official" semver regexp permits such weird strings.